File: Models/ViewModels/BookingVM.cs
using GoWheels_WebAPI.Models.Entities;

namespace GoWheels_WebAPI.Models.ViewModels
{
    public class BookingVM : BaseModelVM
    {
        public decimal PrePayment { get; set; }
        public decimal Total { get; set; }
        public decimal FinalValue { get; set; }
        public DateTime RecieveOn { get; set; }
        public DateTime ReturnOn { get; set; }
        public string? Status { get; set; }
        public bool IsRequest { get; set; }
        public bool IsResponse { get; set; }
        public string? Longitude { get; set; }
        public string? Latitude { get; set; }
        public bool IsRequireDriver { get; set; }
        public bool HasDriver { get; set; }
        public bool IsPay { get; set; }
        public UserVM User { get; set; } = null!;
        public PostVM Post { get; set; } = null!;
        public PromotionVM Promotion { get; set; } = null!;
        public DriverVM Driver { get; set; } = null!;
    }
}

File: FE/src/Model/BookingVM.js
// src/models/BookingVM.js

import PostVM from './PostVM';
import UserVM from './PostVM';
import PromotionVM from './PromotionVM';
import BaseModel from './BaseModelVM';
import DriverVM from './DriverVM';
export default class BookingVM extends BaseModel{
  constructor(data = {}) {
    super(data.id || null, data.createdById, data.createdOn, data.modifiedById, data.modifiedOn, data.isDeleted);
    this.prePayment = data.prePayment || 0;
    this.isRequireDriver = data.isRequireDriver || false;
    this.total = data.total || 0;
    this.finalValue = data.finalValue || 0;
    this.recieveOn = data.recieveOn || null; // Date
    this.returnOn = data.returnOn || null; // Date
    this.status = data.status || '';
    this.isRequest = data.isRequest || false;
    this.isResponse = data.isResponse || false;
    this.longitude = data.longitude || null;
    this.latitude = data.latitude || null;
    this.isPay = data.isPay || false;
    this.user = new UserVM(data.user || {});
    this.post = new PostVM(data.post || {});
    this.promotion = new PromotionVM(data.promotion || {});
    this.driver = new DriverVM(data.driver || {});
  }
}
